code/game/objects/structures/window.dm

```diff
@@ -209,67 +210,97 @@ var/global/wcCommon = pick(list("#379963", "#0d8395", "#58b5c3", "#49e46e", "#8f
 					M.apply_damage(30)
 					hit(75)
 			return
+	if(I.flags & NOBLUDGEON)
+		return
 
-	if(I.flags & NOBLUDGEON) return
+	if(handle_decon(I, user, is_fulltile()))
+		return
 
-	if(istype(I, /obj/item/weapon/screwdriver))
-		if(reinf && state >= 1)
-			state = 3 - state
-			playsound(loc, I.usesound, 75, 1)
-			to_chat(user, (state == 1 ? "<span class='notice'>You have unfastened the window from the frame.</span>" : "<span class='notice'>You have fastened the window to the frame.</span>"))
-		else if(reinf && state == 0)
-			anchored = !anchored
+	if(I.damtype == BRUTE || I.damtype == BURN)
+		user.changeNext_move(CLICK_CD_MELEE)
+		hit(I.force)
+		if(health <= 7)
+			anchored = 0
 			update_nearby_icons()
-			playsound(loc, I.usesound, 75, 1)
-			to_chat(user, (anchored ? "<span class='notice'>You have fastened the frame to the floor.</span>" : "<span class='notice'>You have unfastened the frame from the floor.</span>"))
-		else if(!reinf)
+			step(src, get_dir(user, src))
+	else
+		playsound(loc, 'sound/effects/Glasshit.ogg', 75, 1)
+	..()
+
+/obj/structure/window/proc/handle_decon(obj/item/weapon/W, mob/user, var/takes_time = FALSE)
+	//screwdriver
+	if(isscrewdriver(W))
+		playsound(loc, W.usesound, 75, 1)
+		if(reinf)
+			if(state == 0)
+				if(takes_time)
+					to_chat(user, "<span class='notice'>You begin to [anchored ? "unfasten the frame from" : "fasten the frame to"] the floor.</span>")
+					if(!do_after(user, 20 * W.toolspeed, target = src))
+						return 1
+				anchored = !anchored
+				to_chat(user, (anchored ? "<span class='notice'>You have fastened the frame to the floor.</span>" : "<span class='notice'>You have unfastened the frame from the floor.</span>"))

```

Comment:
Why does this use a completely different syntax for changing the message as compared to L237?